models/repo_language_stats.go

```diff
@@ -144,3 +144,39 @@ func (repo *Repository) UpdateLanguageStats(commitID string, stats map[string]fl
 
 	return sess.Commit()
 }
+
+// CopyLanguageStat Copy originalRepo language stat information to destRepo (use for forked repo)
+func CopyLanguageStat(originalRepo, destRepo *Repository) error {
+	sess := x.NewSession()
+	defer sess.Close()
+	if err := sess.Begin(); err != nil {
+		return err
+	}
+	originalRepoLang := make(LanguageStatList, 0, 6)
+	destRepoLang := make(LanguageStatList, 0, 6)
+	if err := sess.Where("`repo_id` = ?", originalRepo.ID).Desc("`percentage`").Find(&originalRepoLang); err != nil {
+		return err
+	}
+	if len(originalRepoLang) > 0 {
+		originalRepoLang.loadAttributes()
+		for i := range originalRepoLang {
+			destRepoLangItem := new(LanguageStat)
```

Comment:
It's better to the following form, so that it will be filled once.
```go
var destRepoLangItem = LanguageStat{
...
}
```

Or you can

```go
var destRepoLangItem = *(originalRepoLang[i])
destRepoLangItem.ID = 0
destRepoLangItem.RepoID = destRepo.ID
destRepoLangItem.CreatedUnix = timeutil.TimeStampNow()
```

Reply:
Thanks your guidence.

Reply:
Or copy the slice and then itterate throu ut to change parts mentioned lunny mentioned (ID, RepoID, CreatedUnix)
```go
b = make([]T, len(a))
copy(b, a)
```

Reply:
I will try. Thank you.